#!/usr/bin/env python
PACKAGE = "ml_detector"

from dynamic_reconfigure.parameter_generator_catkin import *
import math

gen = ParameterGenerator()

gen.add("adaptive_thresh_win_size_min", int_t, 0, "", 3, 0, None)
gen.add("adaptive_thresh_win_size_max", int_t, 0, "", 23, 0, None)
gen.add("adaptive_thresh_win_size_step", int_t, 0, "", 10, 0, None)
gen.add("adaptive_thresh_constant", int_t, 0, "", 7, 0, None)

gen.add("min_marker_perimeter_rate", double_t, 0, "", 0.1, 0.0, None)
gen.add("max_marker_perimeter_rate", double_t, 0, "", 4.0, 0.0, None)

gen.add("polygonal_approx_accuracy_rate", double_t, 0, "", 0.0075, 0.0, None)
gen.add("min_corner_distance_rate", double_t, 0, "", 0.05, 0.0, None)
gen.add("min_marker_distance_rate", double_t, 0, "", 0.05, 0.0, None)
gen.add("min_distance_to_border", int_t, 0, "", 10, 0, None)

gen.add("marker_border_bits", int_t, 0, "", 1, 0, None)
gen.add("min_otsu_std_dev", double_t, 0, "", 5.0, 0.0, None)
gen.add("perspective_remove_pixel_per_cell", int_t, 0, "", 8, 0, None)
gen.add("perspective_remove_ignored_margin_per_cell", double_t, 0, "", 0.13, 0.0, None)

gen.add("max_erroneous_bits_in_border_rate", int_t, 0, "", 0, 0, None)
gen.add("error_correction_rate", int_t, 0, "", 0, 0, None)

gen.add("corner_refinement_method", int_t, 0, "", 1, 0, None)
gen.add("corner_refinement_win_size", int_t, 0, "", 5, 0, None)
gen.add("corner_refinement_max_iterations", int_t, 0, "", 30, 0, None)
gen.add("corner_refinement_min_accuracy", double_t, 0, "", 0.05, 0.0, None)

exit(gen.generate(PACKAGE, "ml_detector", "DetectorParams"))
